setup.py

```diff
@@ -27,23 +21,32 @@
     author='Jordan Ambra',
     author_email="jordan.ambra@corva.ai",
     url='https://github.com/corva-ai/python-sdk',
-    version=version,
-    classifiers=classifiers,
+    version="0.0.12",
+    classifiers=CLASSIFIERS,
     description='SDK for interacting with Corva',
-    long_description=readme,
+    long_description=README,
     long_description_content_type="text/markdown",
     keywords='corva, sdk',
-    packages=setuptools.find_packages(
-        ".", include=('corva', 'corva.*', 'corva_plugin')
-    ),
+    py_modules=[file.stem for file in pathlib.Path('src').glob('*.py')],
+    packages=setuptools.find_packages("src"),
+    package_dir={"": "src"},
     install_requires=[
         "fakeredis >=1.4.5, <2.0.0",
         "pydantic >=1.7.3, <2.0.0",
         "redis >=3.5.3, <4.0.0",
         "requests >=2.25.0, <3.0.0",
         "requests-mock >=1.8.0, <2.0.0",
     ],
+    extras_require={
```

Comment:
If you want to move away from having a version locked requirements.txt and putting everything into setup.py, you're going to need to start testing this with a Tox matrix. Otherwise, we have no real guaranteed working version that passes tests.

It's always best for development purposes to have packages locked to a specific version, but here with install_requires we want to allow flexibility on customer-facing installs only. It's OK that these concepts are separate.

Reply:
Default python projects need different requirements files, package python projects only need setup.py.

Only setup.py is needed because:

1. We run tests against the package as it will be installed by users, not against whatever the situation in project directory is. Users will never run requirements.txt only setup.py file.
2. If our tests fail, because of some dependency - we are lucky. We found and fixed the error before user encountered it.

Having only setup.py (or pyproject.toml for other packagers) is a usual thing for libraries to do:

- [fastapi](https://github.com/tiangolo/fastapi/blob/master/pyproject.toml)

- [pytest](https://github.com/pytest-dev/pytest)

About tox matrix. We can add that for test completeness in the future.